src/main/java/seedu/address/logic/commands/EditTaskCommand.java

```diff
@@ -71,6 +72,11 @@ public CommandResult execute(Model model) throws CommandException {
 
         Task editedTask = createEditedTask(taskToEdit, editTaskDescriptor, model);
 
+        //if the updated date is not same as original and if date is not today onwards
+        if (!taskToEdit.getDate().equals(editedTask.getDate()) && editedTask.getDate().isBefore(LocalDate.now())) {
```

Comment:
LocalDate.now() changes as time goes by, that was what i am looking for, like it should check it against the local computer date.

I just tried using LocalTime.now(Clock.systemDefaultZone()), but the result is the same as LocalTime.now()

Reply:
Yea, exactly the result is the same. But during testing you can inject a fixed clock, such that `LocalTime.now(this.clock)` will always return a fixed time. Otherwise it's very hard for you to write the test cases. One day it works, the as time goes by, it doesn't. 

See https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/Clock.html#fixed(java.time.Instant,java.time.ZoneId)

Reply:
okay i see, so if thats the case, then we need a setter for the clock or is there a way to manipulate it?

Reply:
solved